spec/acceptance/suites/rpm/01_simp_server_spec.rb

```diff
@@ -0,0 +1,165 @@
+        create_remote_file(master, '/root/simp_conf.yaml', ERB.new(simp_conf_template).result(binding))
+        # require 'pry';binding.pry
+        on(master, 'simp config -a /root/simp_conf.yaml --quiet --skip-safety-save')
+        # Remove the lock file because we've already added the vagrant user stuff
+        on(master, 'rm /root/.simp/simp_bootstrap_start_lock -f')
+      end
+      it 'should provide default hieradata to make beaker happy' do
+        create_remote_file(master, '/etc/puppetlabs/code/environments/simp/hieradata/default.yaml', {
+          'sudo::user_specifications' => {
+            'vagrant_all' => {
+              'user_list' =>  ['vagrant'],
+              'cmnd'      =>  ['ALL'],
+              'passwd'    =>  false,
+            },
+          },
+          'pam::access::users' => {
+            'defaults' => {
+              'origins'    => ['ALL'],
+              'permission' =>  '+'
+            },
+            'vagrant' => nil
+          },
+          'ssh::server::conf::permitrootlogin'    =>  true,
+          'ssh::server::conf::authorizedkeysfile' =>  '.ssh/authorized_keys',
+          # The followign seetings are becuase $server_facts['serverip'] is
+          # incorrect in a beaker/vagrant environment
+          'simp::puppet_server_hosts_entry'       => false,
+          'simp::rsync_stunnel'                   => master_fqdn
+          }.to_yaml
+        )
+      end
+      it 'should enable autosign' do
+        on(master, 'puppet config --section master set autosign true')
+      end
+      it 'should run simp bootstrap' do
+        # this makes me sad but I am unsure of how to fix it for now
+        # on(master, '/sbin/usermod -d /opt/puppetlabs/server/data/puppetserver puppet')
+        # on(master, 'simp bootstrap --no-verbose -u --remove_ssldir > /dev/null', :accept_all_exit_codes => true)
+        on(master, 'simp bootstrap --no-verbose -u --remove_ssldir > /dev/null')
+        # on(master, 'puppet agent -t', :acceptable_exit_codes => [0,2,4,6])
+      end
+      it 'should reboot the host' do
+        master.reboot
+        sleep(240)
+      end
+      it 'should have puppet runs with no changes' do
+        on(master, '/opt/puppetlabs/bin/puppet agent -t', :acceptable_exit_codes => [0,2,4,6])
+        on(master, '/opt/puppetlabs/bin/puppet agent -t', :acceptable_exit_codes => [0] )
+      end
+      it 'should generate agent certs' do
+        togen = []
+        agents.each do |agent|
+          togen << agent.hostname + '.' + domain
+        end
+        create_remote_file(master, '/var/simp/environments/production/FakeCA/togen', togen.join("\n"))
+        on(master, 'cd /var/simp/environments/production/FakeCA; ./gencerts_nopass.sh')
+      end
+    end
+  end
+
+  # context 'classify nodes' do
+  # end
+
+  context 'agents' do
+    agents.each do |agent|
+      it 'should install the agent' do
+        agent.install_package('http://yum.puppetlabs.com/puppetlabs-release-pc1-el-7.noarch.rpm')
```

Comment:
@jeefberkey You can filter based on the client type and install the correct agent. The point is to make sure that all supported clients can be served by the same code from the same server.

We can make this an additional ticket if we need to.